soc/st/stm32/stm32n6x/CMakeLists.txt

```diff
@@ -28,20 +28,52 @@ if(NOT CONFIG_BOOTLOADER_MCUBOOT)
     set(SIGNING_TOOL STM32_SigningTool_CLI)
   endif()
 
-  find_file(SIGNING_TOOL_FIND ${SIGNING_TOOL})
+  find_program(SIGNING_TOOL_FIND ${SIGNING_TOOL})
   if(SIGNING_TOOL_FIND STREQUAL SIGNING_TOOL_FIND-NOTFOUND)
     message(WARNING "
     Signing Image tool (${SIGNING_TOOL}) is not available.
     Signed image will not be generated.
     You won't be able to run application on the board.
     Refer to board documentation for more information")
   else()
-    set_property(GLOBAL APPEND PROPERTY extra_post_build_commands
-      COMMAND ${SIGNING_TOOL}
+    set(SIGNING_TOOL_MIN_VERSION "2.21.0")
+    set(SIGNING_TOOL_VERSION_REGEX "v([0-9]+[.][0-9]+[.][0-9]+)")
+
+    execute_process(
+      COMMAND ${SIGNING_TOOL_FIND} --version
+      OUTPUT_VARIABLE version
+      RESULT_VARIABLE result
+    )
+
+    set(SIGNING_TOOL_ARGS
       -in ${PROJECT_BINARY_DIR}/${CONFIG_KERNEL_BIN_NAME}.bin
       -nk -t fsbl -hv 2.3 --silent
       -o ${PROJECT_BINARY_DIR}/${CONFIG_KERNEL_BIN_NAME}.signed.bin
       -dump ${PROJECT_BINARY_DIR}/${CONFIG_KERNEL_BIN_NAME}.signed.bin
+    )
+
+    if(result EQUAL 0 AND version MATCHES ${SIGNING_TOOL_VERSION_REGEX})
+      set(SIGNING_TOOL_VERSION ${CMAKE_MATCH_1})
+
+      if(SIGNING_TOOL_VERSION VERSION_GREATER_EQUAL SIGNING_TOOL_MIN_VERSION)
+        list(APPEND SIGNING_TOOL_ARGS -align)
+
+        message(STATUS
+        "Found ${SIGNING_TOOL} version v${SIGNING_TOOL_VERSION}, "
+        "appending \"-align\" flag to post-build signing command "
+        "(applicable only with header v2.3 for MCUs, starting from "
+        "version v${SIGNING_TOOL_MIN_VERSION})"
+        )
+      endif()
+    else()
+      message(FATAL_ERROR
+      "Unable to determine ${SIGNING_TOOL} version (expected match to "
+      "${SIGNING_TOOL_VERSION_REGEX} regex)"
+      )
```

Comment:
Not sure if we want to bomb out here or just throw a warning. I figured it's better to have an explicit error than to arbitrarily choose either appending or omitting the `-align` flag in the case where we can't determine the version.